pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,205 @@
+        - `ORGANIZATION_SLUG` can be obtained:
+
+            * From the end of your Buildkite URL, after accessing **Pipelines** in the global navigation of your organization in Buildkite.
+
+            * By running the [List organizations](/docs/apis/rest-api/organizations#list-organizations) REST API query to obtain this value from `slug` in the response. For example:
+
+                ```bash
+                curl - X GET "https://api.buildkite.com/v2/organizations" \
+                  -H "Authorization: Bearer $TOKEN"
+                ```
+        - `PIPELINE_SLUG` can be obtained:
+
+            * From the end of your Buildkite URL, after accessing **Pipelines** in the global navigation of your organization in Buildkite, then accessing the specific pipeline to be specified in the custom trust policy.
+
+            * By running the [List pipelines](/docs/apis/rest-api/pipelines#list-pipelines) REST API query to obtain this value from `slug` in the response from the specific pipeline. For example:
+
+                ```bash
+                curl - X GET "https://api.buildkite.com/v2/organizations/{org.slug}/pipelines" \
+                  -H "Authorization: Bearer $TOKEN"
+                ```
+        - `REF` is usually replaced with `refs/heads/main` to enforce the IAM role's access and use to only the `main` branch, `refs/tags/*` to ensure only tagged releases are able to be deployed, or a wildcard `*` if the IAM role can be accessed and used by all branches.
+        - `BUILD_COMMIT` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+        - `STEP_KEY` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+
+    **Note:** When formulating your _subject_ field's value, you can replace any of the constituent field values above with a wildcard `*` to not set limits on those constituent fields.
+
+    For example, to allow the IAM role to be used for any pipeline in the Buildkite organization `example-org`, when building on any branch or tag, specify a single _subject_ field value of `organization:example-org:*`.
+
+    You can also allow this IAM role to be used with other pipelines, branches, commits and steps by specifying multiple comma-separated values for the `agent.buildkite.com:sub` _subject_ field.
```

Comment:
It's probably worth calling you to the customer that they should **never** omit the organization slug

Reply:
Thanks @patrobinson , my next revision has specified which constituent field values are optional and which are required.

Would it be OK to specify `organization:*` though? (If not, I'll clarify this.)

Reply:
While you technically could specify that any organization can assume the IAM role with that approach, you absolutely **should not**, because that'll remove all the protections we're encouraging people to set here. They should ensure that only organisations that they control can assume the role by explicitly naming them here.

Reply:
Thanks @duckalini , I'll keep the latest guidance as is.

FYI, our docs IA is a bit of a mess and so I'm restructuring the Security section of the Pipelines docs into their own consistent directory and nav hierarchy before moving the OIDC docs into them.